src/components/add-account-dialog/add-account-dialog.js

```diff
@@ -69,7 +69,10 @@ export class AddAccountDialogComponent {
       name: this.accountNameInput.value,
       balance: parseInt(this.balanceNameInput.value),
       type: this.accountType.innerText,
-      currency: this.accountCurrency.innerText
+      currency: this.accountCurrency.innerText,
+      id: Math.random()
```

Comment:
do you really need id? i think accounts should be unique. in this case you can search by name